src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -503,5 +518,18 @@ private async Task InitializeCollectionPropertiesForBuildAsync()
             this.databaseResourceId = this.databaseResourceId ?? await GetDatabaseResourceIdAsync(this.feedDocumentClient, this.feedCollectionLocation).ConfigureAwait(false);
             this.collectionResourceId = this.collectionResourceId ?? await GetCollectionResourceIdAsync(this.feedDocumentClient, this.feedCollectionLocation).ConfigureAwait(false);
         }
+
+        private ICheckpointPartitionProcessorFactory GetPartitionProcessorFactory(string feedCollectionSelfLink)
+        {
+            if (this.partitionProcessorFactory != null)
+            {
+                if (this.checkpointPartitionProcessorFactory != null)
+                    throw new ArgumentException("Only one partition processor factory can be used");
+
+                this.checkpointPartitionProcessorFactory = new CheckpointPartitionProcessorFactoryAdapter(this.partitionProcessorFactory);
```

Comment:
can we assign to local rather than mutating state of "this"? One issue with that is the builder instance is re-used, the user will get an exception that now there are 2 factories.